Optimize file handling in the find module

dminca opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Refactor the way files are handled within the exception block. Use the best practice to operate on an opened file without thinking about closing the file.
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
- find

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```

```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

dminca committed:
Optimize file handling

Use the best practice of opening and doing operations on an opened file

Signed-off-by: Daniel Andrei Minca <mandrei17@gmail.com>

sivel commented on lib/ansible/modules/files/find.py:
```diff
@@ -255,14 +255,12 @@ def contentfilter(fsname, pattern):
-
-        f.close()
+        with open(fsname) as f:
+            prog = re.compile(pattern)
```
We should move the `re.compile` out of the context manager.  Potentially even before the `try`.

Also we should catch invalid regex patterns separate from issues opening the file.

abadger commented on lib/ansible/modules/files/find.py:
```diff
@@ -250,19 +250,22 @@ def sizefilter(st, size):
+    :param fsname:
+    :param pattern:
+    :rtype: bool
+    '''
```
If you're going to add docstrings, note that we use sphinx format for them:
``` python
"""
Filter files which contain the given expression
:arg  fsname: A filename to scan for lines matching a pattern
:arg pattern: The pattern to look for inside of the file.
:rtype: bool
:returns: True if one of the lines in fsname matched the pattern.  Otherwise, False.
"""

dminca committed:
Fix docstring to Sphinx type

- update the docstrings to Sphinx type, as suggested by Toshio
- Move the pattern object assignment outside the context manager, as
  suggested by Matt

Signed-off-by: Daniel Andrei Minca <mandrei17@gmail.com>

dminca commented:
Docstring has been updated :)

abadger commented:
Is the error perhaps because the pattern was None (ie, we need to handle
the if pattern is None: return True. Before compiling the pattern?)

On Tue, Apr 10, 2018, 12:11 AM Daniel Andrei Mincă <notifications@github.com>
wrote:

> Closed #38192 <https://github.com/ansible/ansible/pull/38192>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ansible/ansible/pull/38192#event-1565444441>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAMxWi-2rHEhc-rLD9-kahT9zHxMw97Qks5tnFsxgaJpZM4TD6Bh>
> .
>

dminca committed:
Fix issue with pattern

Tests are failing because the `pattern is None` check is placed after
variable is being initialized.

As a result, first check if pattern is None, then proceed initializing the var with pattern.

Signed-off-by: Daniel Andrei Minca <mandrei17@gmail.com>

dminca commented:
Thanks @abadger , pushed a fix, I wanna see if this one works out :+1:

abadger commented:
Looks good!  Thanks @dminca.  Merged to devel for 2.6.0
